### ISSUE TITLE
Microsoft.AspNetCore.Authentication deprecated

### ISSUE THREAD

**Issue by ebiton-eig** [dotnet/AspNetCore.Docs#29655]
The nuget package Microsoft.AspNetCore.Authentication is deprecated (.NET Core 2)

How can I do the same thing in .NET 7 without this package?

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: ee71441f-471a-89cd-0ef4-9b32b5dd85cf
* Version Independent ID: 0096c365-2f4e-5d21-89a0-c4f10fcb0ed9
* Content: [Authorize with a specific scheme in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/security/authorization/limitingidentitybyscheme?view=aspnetcore-7.0)
* Content Source: [aspnetcore/security/authorization/limitingidentitybyscheme.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/authorization/limitingidentitybyscheme.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by JeremyLikness** [comment]
Hi @ebiton-eig,

I assume it's the .NET Core 2 package that's deprecated, not the library itself. You should be able to include the same package (the newer version that targets ASP.NET core) in your library projects:

https://www.nuget.org/packages/Microsoft.AspNetCore.Authorization

Regards,

Jeremy

**Comment by ebiton-eig** [comment]
Hi!

No, you're confusing Microsoft.AspNetCore.Authentication with Microsoft.AspNetCore.Authorization. The first one simply no longer exists, and was the one that contained AuthenticationHandler, for example.

Regards.

**Comment by JeremyLikness** [comment]
Good point. Shouldn't have answered so quickly. What are you trying to accomplish? We have separate libs for things like managing JWT tokens, managing identity either externally handled (i.e. Azure) or locally via ASP.NET Core Identity. Can you provide a bit more context?

**Comment by ebiton-eig** [comment]
We use a custom scheme to verify authentication between microservices (no-trust). This system is not based on a standard such as JWT or OAuth2.

**Comment by guardrex** [comment]
@ebiton-eig ... See :point_right: https://learn.microsoft.com/en-us/aspnet/core/fundamentals/target-aspnetcore?view=aspnetcore-7.0&tabs=visual-studio#use-the-aspnet-core-shared-framework

Give your lib access to the shared framework using that guidance, and you should be good for using that assembly's API.

@Rick-Anderson ... I'll leave this open in case you had something in mind for work. I'm going to add something to the Blazor class libs article on this, probably just an additional resource cross-link. I'm going to track that on my Blazor tracking issue. I won't be working it off of this issue.

**Comment by guardrex** [comment]
I'm going to close this. We document this for libs, and this article (the one that this issue happened to be opened for) shouldn't make any remarks pertaining to this discussion. Sorry it took so long. I rediscovered this issue while working on updates for this article to resolve other goals for it.
